### PR TITLE
Added random tests for SegmentedExclusiveSum and deleted those cudpp files

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

File: k2/csrc/ragged_test.cu [hunk 1]
```diff
@@ -2559,24 +2559,42 @@ TEST(RaggedShapeOpsTest, RaggedShapeAxis0Splitter) {
...
+                                 //
+                                 0, 5, 11, 18};
+      CheckArrayData(dst, expected);
+
+      // &src.values == dst
+      SegmentedExclusiveSum(src, &src.values);
+      CheckArrayData(src.values, expected);
+    }
+    {
+      // random case, we assume the implementation for cpu is correct and only
```
Comment (csukuangfj) [inline k2/csrc/ragged_test.cu:2582]: For testing, I would suggest using ExclusiveSum to compute the ground truth for each sublist separately.

File: k2/csrc/test_utils.h [hunk 2]
```diff
@@ -53,33 +53,37 @@ bool ApproxEqual(FloatType a, FloatType b, double delta = 0.001) {
 }
 
 template <typename T>
-void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual) {
+void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual,
+                 double abs_error = 0.001) {
```
Comment (csukuangfj) [inline k2/csrc/test_utils.h:57]: `abs_error` is never used.

File: k2/csrc/ragged_test.cu (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/ragged_test.cu:2582]: mm, the underlying cpu are the same, so there's no difference. Anyway, we have tested cpu code with simple case, so I believe it's fine to just compare the result of GPU version and CPU version.

File: k2/csrc/test_utils.h (hunk 2 above)
Comment (qindazhu) [reply k2/csrc/test_utils.h:57]: It's template function, I used it in below float and double version

Comment (csukuangfj) [reply k2/csrc/test_utils.h:57]: template functions support overload.

```cpp
template<typename T>
void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual);

template<typename T>
void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual, T abs_error);
```
should be fine.

Comment (qindazhu) [reply k2/csrc/test_utils.h:57]: mm..thinking about how you will call the two versions, you must use a if-else based on T. for simplity, let's keep the current version...but you remind me to add documentation here to say the parameter just for float and double.

**Pull request merged**